submodules/telemetry/__init__.py

```diff
@@ -1,133 +1,145 @@
+        """
+        Constructor method. Initializes variables
+        :param config: Config variable passed in from core.
+        """
+        self.config = config
+        self.modules = {}
+        self.general_queue = collections.deque()  # initialize global variables
+        self.log_stack = collections.deque()
+        self.err_stack = collections.deque()
+        self.packet_lock = Lock()
+        self.logger = logging.getLogger("TELEMETRY")
+
+    def set_modules(self, modules: {}) -> None:
+        """
+        Sets modules variable
+        :param modules: Maps strings to classes, like {"telemetry": Telemetry()}
+        :return: None
+        """
+        self.modules = modules
+
+    def enqueue(self, message) -> bool:
+        """
+        Enqueue a message onto the general queue, to be processed later by thread decide()
+        :param message: The message to push onto general queue. Must be a log/error class
+        or command (string - must begin with semicolon, see command_ingest's readme)
+        :return True if a valid message was enqueued, false otherwise
+        """
+        if not ((type(message) is str and message[0] == ';') or type(message) is error.Error or type(
+                message) is log.Log):   # check for valid types, error if invalid
+            self.logger.error("Attempted to enqueue invalid message")
+            return False
+        with self.packet_lock:
+            self.general_queue.append(message)  # append to general queue
+            return True
+
+    def dump(self, radio='aprs') -> bool:
+        """
+        Concatenates packets to fit in max_packet_size (defined in config) and send through the radio, removing the
+        packets from the error and log stacks in the process
+        :param radio: Radio to send telemetry through, either "aprs" or "iridium"
+        :return True if anything was sent, false otherwise
+        """
+        squishedpackets = ""
+        retVal = False
+
+        with self.packet_lock:
+            while len(self.log_stack) + len(self.err_stack) > 0:    # while there's stuff to pop off
+                next_packet = (str(self.err_stack[-1]) if len(self.err_stack) > 0 else str(self.log_stack[-1]))   # for the purposes of determining packet length
+                while len(base64.b64encode((squishedpackets + next_packet).encode('ascii'))) < self.config["telemetry"]["max_packet_size"] and len(self.log_stack) + len(self.err_stack) > 0:
+                    if len(self.err_stack) > 0: # prefer error messages over log messages
+                        squishedpackets += str(self.err_stack.pop())
+                    else:
+                        squishedpackets += str(self.log_stack.pop())
+                squishedpackets = base64.b64encode(squishedpackets.encode('ascii'))
+                # print(squishedpackets)
+                self.modules["aprs"].send(squishedpackets) # , radio) #FIXME currently just using APRS, what about radio_output?
```

Comment:
`radio_output` is deprecated. `dump` takes a `radio` parameter, so instead of doing `modules['aprs'].send()` do `modules[radio].send()`

Reply:
Fixed in 149ce11